File: app/src/main/java/io/github/ovso/righttoknow/ui/main/news/NewsNetwork.java
package io.github.ovso.righttoknow.ui.main.news;

import android.content.Context;
import androidx.annotation.StringRes;
import io.github.ovso.righttoknow.R;
import io.github.ovso.righttoknow.framework.network.NetworkHelper;
import io.github.ovso.righttoknow.ui.main.news.model.NewsResult;
import io.reactivex.Single;

/**
 * Created by jaeho on 2017. 12. 20
 */

public class NewsNetwork extends NetworkHelper {
  public NewsNetwork(Context context, String baseUrl) {
    super(context, baseUrl);
  }

  public Single<NewsResult> getNews(@StringRes int resId) {
    String query = context.getString(resId);
    int display = 50;
    int start = 1;
    String sort = context.getString(R.string.api_sort);
    return getMyApi().getNews(query, display, start, sort);
  }
}

File: app/src/main/java/io/github/ovso/righttoknow/ui/main/news/NewsNetwork.kt
package io.github.ovso.righttoknow.ui.main.news

import android.content.Context
import androidx.annotation.StringRes
import io.github.ovso.righttoknow.R
import io.github.ovso.righttoknow.framework.network.NetworkHelper
import io.github.ovso.righttoknow.ui.main.news.model.NewsResult
import io.reactivex.Single

class NewsNetwork(context: Context, baseUrl: String) : NetworkHelper(context, baseUrl) {
  fun getNews(@StringRes resId: Int): Single<NewsResult> {
    val query = context.getString(resId)
    val display = 50
    val start = 1
    val sort = context.getString(R.string.api_sort)
    return myApi.getNews(query, display, start, sort)
  }
}